src/server.ts

```diff
@@ -53,17 +54,36 @@ export class AttendingServer {
             member.roles.cache.find(role => role.name == queue.name) !== undefined)
     }
 
-    async RemoveMemberFromQueues(member: GuildMember): Promise<number> {
+    async RemoveMemberFromQueues(member: GuildMember, interaction_type: string): Promise<number> {
         let queue_count = 0
         await Promise.all(this.queues.map(queue => {
             if (queue.Has(member)) {
                 queue_count++
-                return queue.Remove(member)
+                return queue.Remove(member, interaction_type)
             }
         }))
         return queue_count
     }
 
+    async RemoveMember(queue_name: string, member: GuildMember, interaction_type: string): Promise<void> {
+        const queue = this.queues.find(queue => queue.name == queue_name)
+        if(interaction_type === 'APPLICATION_COMMAND'){
+            if (queue === undefined) {
+                throw new UserError(`There is not a queue with the name ${queue_name}`)
+            }
+            if (!queue.is_open) {
+                throw new UserError(`The queue "${queue_name}" is currently closed.`)
```

Comment:
Students can still be on a queue while it is closed. This actually happens fairly frequently in Matt's classes. We want them to be able to keep their spot in line in-between office hours. Similarly, I see no reason why a student can't leave the queue if its closed.

Reply:
remove those error cases